src/openforms/forms/tests/admin/test_views.py

```diff
@@ -305,3 +308,271 @@ def test_single_import(self, m_import):
             "Form successfully imported!",
             messages[0].message,
         )
+
+
+@disable_admin_mfa()
+class TestMigrationForm(WebTest):
+    csrf_checks = False
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.admin_user = StaffUserFactory(user_permissions=["forms.change_form"])
+
+    def test_not_superuser_cant_access(self):
+        user = UserFactory(is_staff=False, is_superuser=False)
+        ogone_merchant = OgoneMerchantFactory()
+        form_to_migrate = FormFactory(
+            name="Ogone form 1",
+            payment_backend="ogone-legacy",
+            payment_backend_options={"merchant_id": ogone_merchant.pk},
+        )
+
+        redirect = self.app.post(
+            reverse("admin:forms_payment_migration"),
+            {"forms_to_migrate": [form_to_migrate.pk]},
+            user=user,
+        )
+
+        redirect_url = (
+            f"{reverse('admin:login')}?next={reverse('admin:forms_payment_migration')}"
+        )
+        self.assertRedirects(
+            redirect,
+            redirect_url,
+            fetch_redirect_response=False,
+            target_status_code=302,
+        )
+
+        form_to_migrate.refresh_from_db()
+
+        self.assertEqual(form_to_migrate.payment_backend, "ogone-legacy")
+        self.assertEqual(
+            form_to_migrate.payment_backend_options, {"merchant_id": ogone_merchant.pk}
+        )
+
+    def test_form_permissions(self):
+        user = StaffUserFactory(user_permissions=[])
+        ogone_merchant = OgoneMerchantFactory()
+        form_to_migrate = FormFactory(
+            name="Ogone form 1",
+            payment_backend="ogone-legacy",
+            payment_backend_options={"merchant_id": ogone_merchant.pk},
+        )
+
+        response = self.app.post(
+            reverse("admin:forms_payment_migration"),
+            {"forms_to_migrate": [form_to_migrate.pk]},
+            user=user,
+            expect_errors=True,
+        )
+
+        self.assertEqual(response.status_code, 403)
+
+        form_to_migrate.refresh_from_db()
+
+        self.assertEqual(form_to_migrate.payment_backend, "ogone-legacy")
+        self.assertEqual(
+            form_to_migrate.payment_backend_options, {"merchant_id": ogone_merchant.pk}
+        )
+
+    def test_confirmation_page(self):
```

Comment:
what confirmation page? what is being tested here?

Reply:
It tests the confirmation page is rendered properly:
<img width="1871" height="928" alt="image" src="https://github.com/user-attachments/assets/c3e20374-c099-4c79-a348-0e24c1786b4f" />